proposals/add-array-object-feature-option-types.md

```diff
@@ -0,0 +1,170 @@
+# Proposal: Add `array` and `object` Types to Feature Options Schema
+
+## Problem Statement
+
+The current Dev Container Feature Options Schema lacks support for `array` and `object` types. This limitation hinders the ability to configure features that require lists of values (e.g., dependencies, ports) or key-value mappings (e.g., environment variables, custom settings). Addressing this limitation will enhance the flexibility and usability of the schema.
+
+This proposal aims to address [Issue #567](https://github.com/devcontainers/spec/issues/567).
+
+## Proposed Solution
+
+Extend the Dev Container Feature Options Schema by introducing `array` and `object` as valid types for feature options. These new types will enable developers to create more complex and customizable features.
```

Comment:
What types of values would be supported? Maybe limiting it to strings and booleans would keep it simple and aligned with the existing types.